ingestion/src/metadata/ingestion/source/database/database_service.py

```diff
@@ -533,7 +533,7 @@ def mark_tables_as_deleted(self):
             logger.info(
                 f"Mark Deleted Tables set to True. Processing database [{self.context.database.name.__root__}]"
             )
-            if self.source_config.markDeletedTablesFromFilterOnly:
+            if not self.source_config.markAllDeletedTables:
```

Comment:
`markAllDeletedTables` is the default false so this if block will run.
When the value is false that time else block will run.
(Once we ingest  database service without FilterPattern and after we use with FilterPattern, now it will not give 'Deleted' mark in other tabes )

Reply:
Hi @MilanBariya this took me a while to understand, maybe because we are checking a `not bool` condition and there are not comments in the code.

Could we change the `if`s here so that:

```python
if self.source_config.markDeletedTables:
    ...
    if self.source_config.markAllDeletedTables:  # if enabled, delete all
        yield from self.fetch_all_schema_and_delete_tables()
    else:  # if not enabled (default) delete from schema
        schema_names_list = self.get_database_schema_names()
        ....
```

and add proper comments around the logic of each block?

Thanks

Reply:
Sure @pmbrull let me do that.
Thank you.